retrybp/filters.go

```diff
@@ -8,8 +8,16 @@ import (
 	retry "github.com/avast/retry-go"
 
 	"github.com/reddit/baseplate.go/clientpool"
+	"github.com/reddit/baseplate.go/internal/gen-go/reddit/baseplate"
 )
 
+// BaseplateError is implemented by the Error struct in baseplate.thrift.
+type BaseplateError interface {
```

Comment:
consider we only used `GetCode`, it's probably worth unexport this type and only have `GetCode` defined.